src/snowflake/connector/cursor.py

```diff
@@ -1333,14 +1333,29 @@ def fetch_arrow_batches(self) -> Iterator[Table]:
         )
         return self._result_set._fetch_arrow_batches()
 
-    def fetch_arrow_all(self) -> Table | None:
+    @overload
+    def fetch_arrow_all(self, force_return_table: Literal[False] = False) -> None:
```

Comment:
This overloading isn't fully correct, see [this example](https://mypy.readthedocs.io/en/stable/literal_types.html). Also I think instead of `-> None` it should be the old behavior, so `-> Table | None` when `False` is provided.

Reply:
you are right, updated